lazy_theta_star_planner/src/lazy_theta_star2.cpp

```diff
@@ -0,0 +1,257 @@
+//  Copyright 2020 Anshumaan Singh
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http:// www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+#include "lazy_theta_star_planner/lazy_theta_star2.h"
+#include <vector>
+
+namespace lazyThetaStar
+{
+
+LazyThetaStar::LazyThetaStar()
+{
+  sizeX = 0;
+  sizeY = 0;
+}
+
+bool LazyThetaStar::getPath(std::vector<coords<world_pts>> & raw_path)
+{
+  RCLCPP_INFO(node_->get_logger(), "Path Planning Begins....");
+
+  initializeStuff();
+
+  std::cout << src.x << '\t' << src.y << '\t' << dst.x << '\t' << dst.y << '\n';
+
+  data.push_back({src.x, src.y, 0, dist(src.x, src.y, dst.x, dst.y),
+      0, 1, dist(src.x, src.y, dst.x, dst.y)});
+  addIndex(src.x, src.y, id_gen);
+  pq.push({id_gen, &(data[id_gen].f)});
+
+  id_gen++;
+  id curr_id = 0;
+
+  if (!losCheck(src.x, src.y, dst.x, dst.y)) {
+    while (!pq.empty()) {
+
+      tree_node curr_data = data[curr_id];
+      curr_data.closed = -1;
+      id curr_par = curr_data.parent_id;
+
+      if (isGoal(curr_data.x, curr_data.y)) {
+        if (!(losCheck(curr_data.x, curr_data.y, data[curr_par].x, data[curr_par].y))) {
+          resetParent(curr_data);
+        }
+        break;
+      }
+
+      if (!(losCheck(curr_data.x, curr_data.y, data[curr_par].x, data[curr_par].y))) {
+        resetParent(curr_data);
+      }
+
+      setNeighbors(curr_data);
+
+      curr_id = pq.top().pos_id;
+      pq.pop();
+    }
+    if (pq.empty()) {
+      RCLCPP_INFO(node_->get_logger(), "No Path Found !!!!!!!!!!!");
+      raw_path.clear();
+      return false;
+    }
+
+  } else {
+    RCLCPP_INFO(node_->get_logger(), "Straight Line Path Found!");
+    data.push_back({dst.x, dst.y, dist(src.x, src.y, dst.x, dst.y), 0, 0, 1,
+        dist(src.x, src.y, dst.x, dst.y)});
+    curr_id = id_gen;
+  }
+
+  RCLCPP_INFO(node_->get_logger(), "REACHED DEST  %i,   %i", data[curr_id].x, data[curr_id].y);
+
+  backtrace(raw_path, curr_id);
+
+  return true;
+}
+
+void LazyThetaStar::initializePosn()
+{
+  posn.clear();
+  for (int i = 0; i < sizeX * sizeY; i++) {
+    posn.push_back(-1);
+  }
+}
+
+bool LazyThetaStar::losCheck(map_pts & x0, map_pts & y0, map_pts & x1, map_pts & y1) const
+{
+  int dy = abs(y1 - y0), dx = abs(x1 - x0), f = 0;
+  int sx = x1 > x0 ? 1 : -1, sy = y1 > y0 ? 1 : -1;
+  int cx = x0, cy = y0;
+  int ux = (sx - 1) / 2, uy = (sy - 1) / 2;
+
+  if (dx >= dy) {
+    while (cx != x1) {
+      f += dy;
+      if (f >= dx) {
+        if (!isSafe(cx + ux, cy + uy)) {
```

Comment:
You call is safe a bunch of times, maybe you should compute that at the start of the while loop and use the value.